components/sign/ui-sign-input/SignEmailInput.tsx

```diff
@@ -0,0 +1,63 @@
+import styles from "./SignInput.module.scss";
+import classNames from "classnames/bind";
+import { SignEmailInputProps } from "./constant";
+import { EMAIL_REGEX } from "./constant";
+
+const cx = classNames.bind(styles);
+
+export const SignEmailInput = ({
+  values,
+  setValues,
+  emailErrorInputRef,
+  emailErrorMessageRef,
+  showEmailError,
+  setShowEmailError,
+}: SignEmailInputProps) => {
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setValues({ ...values, email: e.target.value });
+  };
+
+  const handleBlur = () => {
+    const isEmailValid = new RegExp(EMAIL_REGEX).test(values.email);
+    setShowEmailError(true);
+
+    if (!values.email) {
+      emailErrorMessageRef.current.textContent = "이메일을 입력해주세요.";
+      return;
+    }
+
+    if (!isEmailValid) {
+      emailErrorMessageRef.current.textContent =
+        "올바른 이메일 주소가 아닙니다.";
+      return;
+    }
+
+    if (values.email && isEmailValid) {
+      return setShowEmailError(false);
+    }
+  };
```

Comment:
```suggestion
const handleBlur = () => {

    const isEmailValid = EMAIL_REGEX.test(values.email);

    let errorMessage = "";
    if (!values.email) {
        errorMessage = "이메일을 입력해주세요.";
    } else if (!isEmailValid) {
        errorMessage = "올바른 이메일 주소가 아닙니다.";
    }

    if (errorMessage) {
        emailErrorMessageRef.current.textContent = errorMessage;
        setShowEmailError(true);
    } else {
        setShowEmailError(false);
    }
};
```

함수안에서 코드 흐름을 살짝 바꿔봤어요.

1. 이메일 유효성 검사하고 
2. 이메일 입력 여부랑 유효성에 따른 에러 메시지를 설정하고
3. 에러 메시지가 있으면 에러를 표시, 없다면 에러 표시를 제거 
하는 흐름으로 살짝 변경해봤습니다